LungMAP prod snapshot trips content type assertion

IT prod failure caused by snapshot `lungmap_prod_0229ea32ef02489eb11eff15819e22c1__20250415_20250521_lm9`:

`[WARNING]	2025-06-26T01:01:51.560Z	8e08cdcc-fd96-53cd-9e15-ad1a63a6db06	azul.indexer.action_controller	Worker failed to handle message {'action': 'add', 'notification': {'transaction_id': '34f7bcab-bce7-4e23-96ff-7fff9e46a702', 'bundle_fqid': {'uuid': '032d0d59-bcb2-3353-9b96-327e514f8e70', 'version': '2025-03-14T14:07:17.783285Z', 'source': {'id': 'ecddba4d-335e-40ef-b08d-04e72ad339c3', 'spec': 'tdr:bigquery:gcp:datarepo-6a2dd798:lungmap_prod_0229ea32ef02489eb11eff15819e22c1__20250415_20250521_lm9:/1'}}}, 'catalog': 'lm9-it'}`
```
Traceback (most recent call last):
  File "/var/task/azul/indexer/action_controller.py", line 63, in _handle_events
    message_handler(message)
  File "/var/task/azul/indexer/index_controller.py", line 146, in _contribute
    contributions, replicas = self.transform(catalog, notification, delete)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/indexer/index_controller.py", line 189, in transform
    bundle = service.fetch_bundle(catalog, bundle_fqid)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/indexer/index_service.py", line 198, in fetch_bundle
    return plugin.fetch_bundle(bundle_fqid)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/plugins/repository/tdr.py", line 205, in fetch_bundle
    bundle = self._emulate_bundle(bundle_fqid)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/plugins/repository/tdr_hca/__init__.py", line 332, in _emulate_bundle
    bundle.add_entity(entity=entity,
  File "/var/task/azul/plugins/repository/tdr_hca/__init__.py", line 163, in add_entity
    self._add_manifest_entry(entity, self.file_from_row(row))
                                     ^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/plugins/repository/tdr_hca/__init__.py", line 191, in file_from_row
    return HCAFile.from_descriptor(descriptor,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/plugins/metadata/hca/__init__.py", line 505, in from_descriptor
    assert ';' not in content_type, R(
           ^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: R('Unexpected MIME parameter in content type', 'text/plain; charset=utf-8')
```

Which is in assertion [introduced](https://github.com/DataBiosphere/azul/pull/7235/commits/ba420fdb01f9deb078ca1928a324c25d0c0b3b97#diff-4cb77695788ed1c372d3d107b43b14170fcc4a86a1ef93e444d2932216619794R503-R504) as part of this promotion.

_Originally posted by @achave11-ucsc in https://github.com/DataBiosphere/azul/issues/7235#issuecomment-3006705965_

--- a/lambdas/indexer/app.py
+++ b/lambdas/indexer/app.py
@@ -6,7 +6,6 @@ from typing import (
 import chalice
 
 from azul import (
-    JSON,
     cached_property,
     config,
 )
@@ -39,6 +38,7 @@ from azul.openapi import (
     format_description as fd,
 )
 from azul.types import (
+    JSON,
     not_none,
 )
 

--- a/scripts/zenhub_to_github.py
+++ b/scripts/zenhub_to_github.py
@@ -35,7 +35,6 @@ from more_itertools import (
 )
 
 from azul import (
-    JSON,
     cached_property,
     format_description as fd,
 )
@@ -50,6 +49,7 @@ from azul.logging import (
 )
 from azul.types import (
     AnyJSON,
+    JSON,
     JSONs,
     PrimitiveJSON,
     json_bool,

--- a/src/azul/indexer/index_repository_service.py
+++ b/src/azul/indexer/index_repository_service.py
@@ -8,7 +8,6 @@ from typing import (
 
 from azul import (
     CatalogName,
-    JSON,
     cache,
 )
 from azul.indexer import (
@@ -18,6 +17,9 @@ from azul.indexer import (
 )
 from azul.plugins import (
     RepositoryPlugin,
+)
+from azul.types import (
+    JSON,
 )
 
 log = logging.getLogger(__name__)

--- a/src/azul/indexer/mirror_service.py
+++ b/src/azul/indexer/mirror_service.py
@@ -19,7 +19,6 @@ from furl import (
 
 from azul import (
     CatalogName,
-    JSON,
     R,
     cached_property,
     config,
@@ -52,6 +51,9 @@ from azul.plugins import (
 from azul.service.storage_service import (
     StorageObjectNotFound,
     StorageService,
+)
+from azul.types import (
+    JSON,
 )
 
 if TYPE_CHECKING:

--- a/src/azul/plugins/metadata/hca/__init__.py
+++ b/src/azul/plugins/metadata/hca/__init__.py
@@ -10,7 +10,6 @@ from attrs import (
 )
 
 from azul import (
-    JSON,
     R,
     config,
     iif,
@@ -65,6 +64,7 @@ from azul.service.manifest_service import (
     ManifestFormat,
 )
 from azul.types import (
+    JSON,
     MutableJSON,
     json_dict,
     json_dict_of_dicts,

--- a/src/azul/plugins/metadata/hca/__init__.py
+++ b/src/azul/plugins/metadata/hca/__init__.py
@@ -10,6 +10,7 @@ from attrs import (
 )
 
 from azul import (
+    CatalogName,
     R,
     config,
     iif,
@@ -525,6 +526,7 @@ class HCAFile(File):
     @classmethod
     def from_metadata(cls,
                       *,
+                      catalog: CatalogName,
                       metadata: JSON,
                       descriptor: JSON,
                       drs_uri: str | None
@@ -539,10 +541,10 @@ class HCAFile(File):
         parameter_suffix = '; dcp-type=data'
         if content_type.endswith(parameter_suffix):
             content_type = content_type.removesuffix(parameter_suffix)
+        elif config.catalogs[catalog].atlas == 'lungmap':
+            pass
         else:
-            # FIXME: Re-enable assertion, potentially in a weakened form
-            #        https://github.com/DataBiosphere/azul/issues/7244
-            assert True or ';' not in content_type, R(
+            assert ';' not in content_type, R(
                 'Unexpected MIME parameter in content type', content_type)
         return cls(uuid=json_str(descriptor['file_id']),
                    name=json_str(json_mapping(metadata['file_core'])['file_name']),

--- a/src/azul/plugins/metadata/hca/__init__.py
+++ b/src/azul/plugins/metadata/hca/__init__.py
@@ -542,6 +542,8 @@ class HCAFile(File):
         if content_type.endswith(parameter_suffix):
             content_type = content_type.removesuffix(parameter_suffix)
         elif config.catalogs[catalog].atlas == 'lungmap':
+            # FIXME: Re-enable content-type validation for lungmap
+            #        https://github.com/DataBiosphere/azul/issues/7244
             pass
         else:
             assert ';' not in content_type, R(

--- a/src/azul/plugins/repository/canned/__init__.py
+++ b/src/azul/plugins/repository/canned/__init__.py
@@ -209,7 +209,8 @@ class Plugin(RepositoryPlugin[
         assert prefix == prefix.lower(), prefix
         staging_area = self.staging_area(source.spec.name)
         return [
-            HCAFile.from_metadata(metadata=staging_area.metadata[file_uuid].content,
+            HCAFile.from_metadata(catalog=self.catalog,
+                                  metadata=staging_area.metadata[file_uuid].content,
                                   descriptor=descriptor.content,
                                   drs_uri=None)
             for file_uuid, descriptor in staging_area.descriptors.items()

--- a/src/azul/plugins/repository/tdr_hca/__init__.py
+++ b/src/azul/plugins/repository/tdr_hca/__init__.py
@@ -279,7 +279,8 @@ class Plugin(TDRPlugin[TDRHCABundle, TDRBundleFQID]):
         bundle.metadata[str(entity)] = content
 
     def _file_from_row(self, row: BigQueryRow) -> HCAFile:
-        return HCAFile.from_metadata(metadata=json.loads(any_str(row['content'])),
+        return HCAFile.from_metadata(catalog=self.catalog,
+                                     metadata=json.loads(any_str(row['content'])),
                                      descriptor=json.loads(any_str(row['descriptor'])),
                                      drs_uri=optional(any_str, row['file_id']))
 

--- a/src/azul/schemas.py
+++ b/src/azul/schemas.py
@@ -4,7 +4,6 @@ from typing import (
 )
 
 from azul import (
-    JSON,
     format_description as fd,
     mutable_furl,
 )
@@ -15,6 +14,9 @@ from azul.openapi import (
     params,
     responses,
     schema,
+)
+from azul.types import (
+    JSON,
 )
 
 

--- a/terraform/cloudwatch_dashboard.template.json.py
+++ b/terraform/cloudwatch_dashboard.template.json.py
@@ -7,13 +7,15 @@ from more_itertools import (
 )
 
 from azul import (
-    JSON,
     R,
     config,
     iif,
 )
 from azul.deployment import (
     aws,
+)
+from azul.types import (
+    JSON,
 )
 
 es_instance_count = (

--- a/test/indexer/test_notifications.py
+++ b/test/indexer/test_notifications.py
@@ -13,14 +13,14 @@ import requests
 from app_test_case import (
     LocalAppTestCase,
 )
-from azul import (
-    JSON,
-)
 from azul.deployment import (
     aws,
 )
 from azul.hmac import (
     SignatureHelper,
+)
+from azul.types import (
+    JSON,
 )
 from azul_test_case import (
     DCP1TestCase,

--- a/test/service/test_app_logging.py
+++ b/test/service/test_app_logging.py
@@ -15,7 +15,6 @@ import requests
 
 from azul import (
     Config,
-    JSON,
 )
 from azul.chalice import (
     AzulChaliceApp,
@@ -23,6 +22,9 @@ from azul.chalice import (
 )
 from azul.logging import (
     configure_test_logging,
+)
+from azul.types import (
+    JSON,
 )
 from indexer import (
     DCP1CannedBundleTestCase,

--- a/test/service/test_response_anvil.py
+++ b/test/service/test_response_anvil.py
@@ -1,13 +1,13 @@
 import requests
 
-from azul import (
-    JSON,
-)
 from azul.logging import (
     configure_test_logging,
 )
 from azul.plugins.repository.tdr_anvil import (
     TDRAnvilBundleFQID,
+)
+from azul.types import (
+    JSON,
 )
 from indexer.test_anvil import (
     AnvilIndexerTestCase,

--- a/test/test_openapi.py
+++ b/test/test_openapi.py
@@ -7,7 +7,6 @@ from furl import (
 )
 
 from azul import (
-    JSON,
     RequirementError,
 )
 from azul.chalice import (
@@ -22,6 +21,9 @@ from azul.logging import (
 from azul.openapi import (
     params,
     schema,
+)
+from azul.types import (
+    JSON,
 )
 from azul_test_case import (
     AzulUnitTestCase,

--- a/test/test_tagging.py
+++ b/test/test_tagging.py
@@ -1,11 +1,11 @@
-from azul import (
-    JSON,
-)
 from azul.logging import (
     configure_test_logging,
 )
 from azul.terraform import (
     _transform_tf,
+)
+from azul.types import (
+    JSON,
 )
 from test.azul_test_case import (
     AzulUnitTestCase,

